lib/rules/curly.js

```diff
@@ -309,6 +309,98 @@ module.exports = {
                 hasUnsafeIf(statement) && isFollowedByElseKeyword(node);
         }
 
+        /**
+         * Examples:
+         *
+         * The following location should be reported
+         *
+         * if (foo) bar()
+         *         ^
+         *
+         * while(foo) bar();
+         *           ^
+         *
+         * for(;;) bar();
+         *         ^
+         *
+         * for (var foo of bar) console.log(foo)
+         *                      ^
+         *
+         * for (var foo in bar) console.log(foo)
+         *                      ^
+         *
+         * Get the location of missing parent location where it should have been placed.
+         * @param {ASTNode} node the node to check.
+         * @param {string} name name to check.
+         * @returns {Object} Position object.
+         */
+        function getMissingLoc(node, name) {
+            switch (name) {
+                case "else":
+                    return getElseKeyword(node).loc.start;
+                case "if":
+                    return astUtils.getNextLocation(sourceCode, node.test.loc.end);
+                case "while":
+                    return astUtils.getNextLocation(sourceCode, node.test.loc.end);
+                case "for":
+                    return node.body.loc.start;
+                case "for-of":
+                    return node.body.loc.start;
+                case "do":
+                    return node.body.loc.start;
+                case "for-in":
+                    return node.body.loc.start;
+                default:
+                    return node.loc;
+            }
+        }
+
+        /**
+         * Examples:
+         *
+         * The following location should be reported
+         *
+         * if (true) foo(); else { baz(); }
+         *                       ^
+         *
+         * for (;;) { foo(); }
+         *          ^
+         *
+         * while (bar) { foo(); }
+         *             ^
+         *
+         * do{foo();} while(bar);
+         *   ^
+         *
+         * for (var foo of bar) {console.log(foo)}
+         *                      ^
+         *
+         * Get the location of unexpected parens. mostly the starting parens `{`
```

Comment:
Can we change `parens` to `curly braces` in both the JSDoc blocks and the function name?